torch/_inductor/utils.py

```diff
@@ -1640,8 +1658,14 @@ def aoti_compile_with_persistent_cache(
                 else:
                     meta_info_item["device_index"] = f"{input_tensor.device.index}"
                 meta_info_item["dtype"] = f"{input_tensor.dtype}"
-                meta_info_item["sizes"] = f"{list(input_tensor.size())}"
-                meta_info_item["strides"] = f"{list(input_tensor.stride())}"
+                if input_tensor.ndim == 0:
+                    # Scalar tensor
+                    meta_info_item["sizes"] = "[]"
+                    meta_info_item["strides"] = "[]"
+                    meta_info_item["scalar_value"] = f"{input_tensor.item()}"
```

Comment:
Don't we only want to do this if the original value came from an int/float/bool?  If the user provides a 0D tensor I would expect it not to specialize.

Reply:
Make sense. We should not specialize 0D tensor. I will refine the PR. Thanks for the comments.